### PR TITLE
Apo

### PR DESCRIPTION
**Pull request by ABDALLAHKAJJ**
IN-1:Validation Eklemek

### CODE REVIEW

**Review by ABDALLAHKAJJ — COMMENTED** [review]
IN-2: Tab Title değişecek

File: src/app/InventoryApp/Stock/product-manager/product-manager.component.html [hunk 1]
```diff
@@ -28,60 +28,90 @@ <h2 class="m-0">Full width with content scroll</h2>
...
                                 {{ Gender.ViewValue }}
                             </mat-option>
+                            <mat-error *ngIf="hasError('Gender', 'required')">
+                                {{'VALIDATIONS.CONTROL_REQURIRED' | translate:{formName:_translate.instant('STOCK_MODULE.MASTER_DATA.GENDER')} }}
+                            </mat-error>
                         </mat-select>
                     </mat-form-field>
                     <mat-form-field class="ml-8 custom-css">
-                        <input type="text" minlength="4" maxlength="4"
+                        <input type="text" minlength="4" maxlength="4" pattern="^(19|20)\d\d$"
```
Comment (foaudkajj) [inline src/app/InventoryApp/Stock/product-manager/product-manager.component.html:63]: What is the propose of `pattern="^(19|20)\d\d$"` ?

File: src/app/InventoryApp/Stock/product-manager/product-manager.component.html [hunk 2]
```diff
@@ -28,60 +28,90 @@ <h2 class="m-0">Full width with content scroll</h2>
...
+                                {{'VALIDATIONS.CONTROL_REQURIRED' | translate:{formName:_translate.instant('STOCK_MODULE.MASTER_DATA.PRODUCT_CODE')} }}
+                            </mat-error>
                         <!-- <mat-hint align="start">Please enter your full name</mat-hint> -->
                     </mat-form-field>
 
 
                     <mat-form-field class="ml-8 custom-css">
                         <mat-select formControlName="Color" autocomplete="off"
-                            placeholder="{{_translate.instant('STOCK_MODULE.MASTER_DATA.COLOR_NAME')}}">
+                            placeholder="{{_translate.instant('STOCK_MODULE.MASTER_DATA.COLOR_NAME')}}" required>
```
Comment (foaudkajj) [inline src/app/InventoryApp/Stock/product-manager/product-manager.component.html:89]: `required ` has been already added in the code:
```
      Color: ['', Validators.compose([
        Validators.required,
        Validators.maxLength(4)
      ])],
```

File: src/app/InventoryApp/Stock/product-manager/product-manager.component.html (hunk 1 above)
Comment (ABDALLAHKAJJ) [reply src/app/InventoryApp/Stock/product-manager/product-manager.component.html:63]: it was used to assure that the date is for this century or the previous century.

**Review by foaudkajj — APPROVED** [review]
Generally, well done.

File: src/app/InventoryApp/Stock/product-manager/product-manager.component.html (hunk 2 above)
Comment (ABDALLAHKAJJ) [reply src/app/InventoryApp/Stock/product-manager/product-manager.component.html:89]: ok, it will be edited out.

**Comment by ABDALLAHKAJJ** [comment]
> Generally, well done.

10x

**Pull request merged**